opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -46,95 +62,179 @@ def __init__(
+    ) -> APIHistogram:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_histogram(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_gauge(
         self, name, callback, unit=None, description=None
-    ) -> ObservableGauge:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableGauge:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_gauge(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_up_down_counter(
         self, name, callback, unit=None, description=None
-    ) -> ObservableUpDownCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableUpDownCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_up_down_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
 
 class MeterProvider(APIMeterProvider):
     """See `opentelemetry._metrics.MeterProvider`."""
 
     def __init__(
         self,
+        metric_exporters: Sequence[MetricExporter] = (),
+        metric_readers: Sequence[MetricReader] = (),
+        views: Sequence[View] = (),
         resource: Resource = Resource.create({}),
         shutdown_on_exit: bool = True,
+        use_always_matching_view: bool = True,
     ):
-        self._resource = resource
+        self._lock = Lock()
         self._atexit_handler = None
+        self.__use_always_matching_view = use_always_matching_view
 
         if shutdown_on_exit:
             self._atexit_handler = register(self.shutdown)
 
-        self._metric_readers = []
-        self._metric_exporters = []
-        self._views = []
-        self._shutdown = False
+        self.__metric_readers = metric_readers
 
-    def get_meter(
-        self,
-        name: str,
-        version: Optional[str] = None,
-        schema_url: Optional[str] = None,
-    ) -> Meter:
+        for metric_reader in self._metric_readers:
+            metric_reader._register_meter_provider(self)
 
-        if self._shutdown:
-            _logger.warning(
-                "A shutdown `MeterProvider` can not provide a `Meter`"
-            )
-            return _DefaultMeter(name, version=version, schema_url=schema_url)
+        self.__metric_exporters = metric_exporters
 
-        return Meter(InstrumentationInfo(name, version, schema_url), self)
+        self.__views = views
 
-    def shutdown(self):
-        # FIXME implement a timeout
+        if self.__use_always_matching_view:
+            self.__views = [*self.__views, View(instrument_name="*")]
 
-        if self._shutdown:
-            _logger.warning("shutdown can only be called once")
-            return False
+        self.__resource = resource
+        self._shutdown = False
+        self.__synchronous_instruments = []
+        self.__asynchronous_instruments = []
```

Comment:
Why do we need these lists? They look unused in this PR

Reply:
The first one is actually unused, I have removed it, the second will be used when collecting.

Reply:
Edit: I am removing both lists as the time being, since they are used by the `MeasurementConsumer` which goes in a later PR.